### PR TITLE
class-generator: do not collect deprecated resources

### PR DESCRIPTION
**Pull request by myakove**
## Summary by CodeRabbit

- **New Features**
  - Introduced various new resource types related to Kubernetes, including `ProvisioningList`, `SchedulerList`, and `OperatorCondition`.
  - Added multiple entries for managing Ceph-related configurations in the schema, enhancing integration with Ceph storage solutions.

- **Refactor**
  - Updated resource management functions for improved clarity and performance in handling Kubernetes resources.

- **Bug Fixes**
  - Streamlined logic in resource handling functions, enhancing overall performance and reducing complexity.

- **Documentation**
  - Enhanced JSON output formatting for better readability, with added indentation for generated mapping files.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: class_generator/class_generator.py [hunk 1]
```diff
@@ -75,29 +73,34 @@ def map_kind_to_namespaced(client: str):
...
             not_kind_list.append(_res["kind"])
 
     with open(RESOURCES_MAPPING_FILE, "w") as fd:
-        json.dump(resources_mapping, fd)
+        json.dump(resources_mapping, fd, indent=4)
 
     with open(not_kind_file, "w") as fd:
         fd.writelines("\n".join(not_kind_list))
 
+    __import__("ipdb").set_trace()
```
Comment (rnetser) [inline class_generator/class_generator.py:102]: leftover?

File: class_generator/tests/manifests/Machine/machine.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # Generated using https://github.com/RedHatQE/openshift-python-wrapper/blob/main/scripts/resource/README.md
 
 from typing import Any, Dict, List, Optional
-from ocp_resources.resource import NamespacedResource
+from ocp_resources.resource import Resource
 
 
-class Machine(NamespacedResource):
+class Machine(Resource):
```
Comment (rnetser) [inline class_generator/tests/manifests/Machine/machine.py:7]: `Machine` is  `NamespacedResource`

File: class_generator/tests/manifests/Secret/secret.py [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Generated using https://github.com/RedHatQE/openshift-python-wrapper/blob/main/scripts/resource/README.md
 
 from typing import Any, Dict, Optional
-from ocp_resources.resource import NamespacedResource
+from ocp_resources.resource import Resource
 
 
-class Secret(NamespacedResource):
+class Secret(Resource):
```
Comment (rnetser) [inline class_generator/tests/manifests/Secret/secret.py:7]: ditto

Comment (myakove) [reply class_generator/tests/manifests/Secret/secret.py:7]: will fix

File: class_generator/tests/manifests/Machine/machine.py (hunk 2 above)
Comment (myakove) [reply class_generator/tests/manifests/Machine/machine.py:7]: Yes

File: class_generator/tests/manifests/Pod/pod.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
 # Generated using https://github.com/RedHatQE/openshift-python-wrapper/blob/main/scripts/resource/README.md
 
 from typing import Any, Dict, List, Optional
-from ocp_resources.resource import NamespacedResource, MissingRequiredArgumentError
+from ocp_resources.resource import Resource, MissingRequiredArgumentError
 
 
-class Pod(NamespacedResource):
+class Pod(Resource):
```
Comment (myakove) [inline class_generator/tests/manifests/Pod/pod.py:7]: Pod is namespaced.

**Review by rnetser — DISMISSED** [review]

File: class_generator/class_generator.py [hunk 5]
```diff
@@ -71,37 +87,45 @@ def map_kind_to_namespaced(client: str):
...
+            if _key in not_kind_list:
                 continue
 
-            _kind_data_futures.append(executor.submit(_is_kind_and_namespaced, client=client, _data=_data))
+            _kind_data_futures.append(executor.submit(_is_kind_and_namespaced, client=client, _key=_key, _data=_data))
 
+    _temp_resources_mappings: Dict[Any, Any] = {}
     for res in as_completed(_kind_data_futures):
         _res = res.result()
+        # _res["kind"] is group.version.kind, set only kind as key in the final dict
```
Comment (dbasunag) [inline class_generator/class_generator.py:105]: Do you want to remove it?

**Pull request merged**